### PR TITLE
PIN-4903 BKE - Error during selfcare information retrieve at the creation of agreement contract

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala [hunk 1]
```diff
@@ -119,14 +124,26 @@ final class AgreementContractCreator(
...
+      submission                 <- seed.stamps.submission.toFuture(StampNotFound("submission"))
+      consumerSelfcareId         <- consumer.selfcareId.toFuture(SelfcareIdNotFound(consumer.id))
+      producerSelfcareId         <- producer.selfcareId.toFuture(SelfcareIdNotFound(producer.id))
+      consumerSelfcareUuid       <- consumerSelfcareId.toFutureUUID
+      producerSelfcareUuid       <- producerSelfcareId.toFutureUUID
+      consumerSelfcare           <- selfcareV2ClientService.getInstitution(consumerSelfcareUuid).map(_.toApi)
+      producerSelfcare           <- selfcareV2ClientService.getInstitution(producerSelfcareUuid).map(_.toApi)
+      (consumerApi, producerApi) <- consumerSelfcare.toFuture.zip(producerSelfcare.toFuture)
+      userResponse               <- selfcareV2ClientService
+        .getUserById(consumerSelfcareUuid, consumerApi.id)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala]: the `consumerApi.id` is exactly the `consumerSelfcareId`, no need to retrieve it again.
Same for `producerApi.id`.

Plus, the second parameter of `getUserById` should be a user id, but here we are passing an institution if

Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala]: I updated the ticket, the submission can be performed only by the consumer, it is enough to retrieve from selfcare the user info always in relation to the consumer

**Review by galales — CHANGES_REQUESTED** [review]
reminder: check tests

File: src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala (hunk 1 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala]: Where I can retrieve userId for consumer?

Comment (galales) [reply src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementContractCreator.scala]: the userId is the `who` field in the stamp

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**